### ISSUE TITLE
SUMM roadmap 0.14 josef

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#7720]
Just trying to get a rough overview what I'm planning for 0.14

- diagnostics, score tests, missing pieces in discrete (avoiding PRs that get too large or widespread)
- almost finished PRs from last year's updating
- add new models
  - heckman
  - Tobit, generic exogenously censored/truncated ?
  - MVGenExtremeModel ? Copula regression model
  - accelerated failure time ?
  - ...
- stats ?
- Refactoring ?
- GMM ?

**Comment by josef-pkt** [comment]
maybe get started with basic dispersion models, Gaussian with heteroscedasticity MLE instead of GLS, negbin with dispersion link function.
(Beta-Binomial ?)
similar to BetaRegression

**Comment by josef-pkt** [comment]
I was looking at `regression.dimred`
It's not advertised and looks like it's mainly for experts (and I don't know anything about it)

Maybe there is a way to make it easier to consume and advertise.
Looking at a similar python package https://github.com/joshloyal/sliced , it's not clear how to interpret it after looking at it only for a short time.

My guess is that we need more usage examples.
E.g. can we use it to predict (or get_distribution) for Poisson and binary regression?

**Comment by josef-pkt** [comment]
open: compare_score_test, compare_lr_test, ...

linear_model has compare_lm_test and lr test

**Comment by josef-pkt** [comment]
more details and design questions

- `get_distribution`  (GLM has now 3 methods for it, results, model and family)
  - from "outside" creating simulation model
  - in sample from results, e.g. for diagnostics 
  - out of sample from results, e.g. for reusing in get_prediction (needs to handle extra args like offset)
- `get_diagnostic`
  - this is mainly to figure out what is available for different models or categories of models
  - and then try to reuse generic code.
  - `score_test` versus some CMT (that don't use explicit model score, e.g. poisson doesn't have parameter for dispersion)
- `get_prediction`
  - need to decide on structure mainly between endpoint transformation and generic delta method  
  - location of helper functions ? e.g. "predict" functions that we can feed to NonlinearDeltaCov
  - families versus dfamilies: 2 different names for model attribute makes generic code more difficult. still undecided
  - overlap with margins, margins also has predict_at exog generation and options
- `get_influence`
  -  not my immediate target
 

semi-random thought: freq_weights_predict in get_prediction
Assume we want to compute population weighted average prediction. i.e. predict weights differ from sample freq_weights.
e.g. ATE versus ATT or ATE for a specific group or subset.

**Comment by josef-pkt** [comment]
diagnostics will be getting large, maybe a subdirectory would be helpful
But I'm still not sure at what class/model hierarchy level we get useful methods.

We will get quite a bit of overlap between genmod, discrete and othermod. and I have not yet looked at other models like survival (proportional hazard, mixedlm, ...)

**Comment by josef-pkt** [comment]
I want to get the multi-link Hetmodels #7793 merged as WIP so I can use them for diagnostic and other post-estimation results. 
We need enhanced `.score_test` for second moment, heteroscedasticity and dispersion tests.
Designing api for `dfamilies` also will need several rounds and use cases for the design.

e.g. 
GaussianHet for MLE based specification tests, instead of current least squares LM test.
NegativeBinomial Poisson special case does not work, but we can have "excess heteroscedasticity" test (dispersion parameter depends on some exog)
GPP: can have Poisson as special case (AFAIR, so we can have MLE score test for Poisson directly.

Problem: we might have to use some numdiff derivatives if we don't have analytical derivatives

**Comment by josef-pkt** [comment]
todo after #7929

- get_influence with resid_pearson and separate score residuals (full quadratic, darg-quadratic, scaled score_obs, and scaled (?) score_factor)
- get_diagnostic: we need at least a basic diagnostic class for binary models
- GLM: modernize prediction, `predict`, `get_prediction` depending on use of `which`. deprecate `linear` keyword. and prediction results class for GLM

**Comment by josef-pkt** [comment]
I'm starting to tend towards adding a `statsmodels.postestimation` directory, where we can add new supporting code and eventually move and consolidate diagnostic, influence and ...(?) modules.

**Comment by josef-pkt** [comment]
A pep-8 cleanup in `discrete` would be useful when we don't have any open PRs for it.
With open PRs, any cleaning causes a huge pain in merge conflicts.

**Comment by josef-pkt** [comment]
smaller PRs to take a break from big design decisions

stats

- #3674 
- #6676 exact 2-proportions
  in private module, need to check if there are better unit tests against R
- #3576  multinomial permutation test, conditional resampling
  one public function in contingency, private randomization module
- #2932 multinomial
  AFAIR, written as a more generic version of multinomial confint, but I ran into problems, direct implementation was merged 
- ..

models and related

- #1791
- #5382
- ..

**Comment by josef-pkt** [comment]
2 smaller enhancements

- generalized logit link class, ELL, Prasetyo et al 2020 (but application is bayesian)
- SJ test, normal versus heavy-tail alternatives, Gel Miao and Gastwirth, used by minitab for quality of variance bonett #8261

**Comment by josef-pkt** [comment]
there are several recent PRs to review and merge

- #8155
- #8317
- #8301 
- #7777
- ...

and I need to check older open PRs (not min) and issues to see what should still go into 0.14

(my new notebook computer doesn't have a working dev environment yet, latest winpython has broken pytest, 
vacation coming soon - post-covid)

**Comment by josef-pkt** [comment]
after a few cleanup/refactoring, doc and notebook merges:

todo, or to look at again and decide

- signature of treatment effect class #8327
- GLM predict `which`
- method="auto" in poisson stats ?
- ...

**Comment by josef-pkt** [comment]
for release checking and notes: new link classes
#8155 logc link merged
other new ?

**Comment by josef-pkt** [comment]
MQuantileNorm is not in docs

**Comment by josef-pkt** [comment]
for release notes we need a list of type-bug-wrong for serious bugs in previous release
e.g. 
- #8425

**Comment by josef-pkt** [comment]
still possible

- [ ] #8364

**Comment by josef-pkt** [comment]
pretty much done here

The only one I'd like to get in is making GLM link check optional.

Also some recently added warnings might use UserWarning which only warns once, in contrast to our own Warning which warn always. needs checking.  
e.g. we have `ValueWarning` as alternative

**Comment by josef-pkt** [comment]
checking docstrings

- MQuantileNorm does not have full references, just author names, formulas are merged into one line
